tests/arch/arm/arm_irq_vector_table/src/arm_irq_vector_table.c

```diff
@@ -16,23 +16,14 @@
  */
 #define _ISR_OFFSET 0
 
-#if defined(CONFIG_SOC_SERIES_NRF51X) || defined(CONFIG_SOC_SERIES_NRF52X)
-/* The customized solution for nRF51X-based and nRF52X-based
- * platforms requires that the POWER_CLOCK_IRQn line equals 0.
- */
-BUILD_ASSERT(POWER_CLOCK_IRQn == 0,
-	"POWER_CLOCK_IRQn != 0. Consider rework manual vector table.");
-
-/* The customized solution for nRF51X-based and nRF52X-based
- * platforms requires that the RTC1 IRQ line equals 17.
- */
-BUILD_ASSERT(RTC1_IRQn == 17,
-	     "RTC1_IRQn != 17. Consider rework manual vector table.");
-
+#if defined(CONFIG_SOC_FAMILY_NRF)
 #undef _ISR_OFFSET
+#if defined(CONFIG_SOC_SERIES_NRF51X) || defined(CONFIG_SOC_SERIES_NRF52X) || \
+	defined(CONFIG_SOC_SERIES_NRF53X) || defined(CONFIG_SOC_SERIES_NRF91X)
+
 #if !defined(CONFIG_BOARD_QEMU_CORTEX_M0)
 /* Interrupt line 0 is used by POWER_CLOCK */
-#define _ISR_OFFSET 1
+#define _ISR_OFFSET TIMER0_IRQn
```

Comment:
The above comment does not make much sense after the change.
How about simplifying this part a bit more and using something like this:
```
#if defined(CONFIG_SOC_FAMILY_NRF)
#undef _ISR_OFFSET
#if defined(CONFIG_BOARD_QEMU_CORTEX_M0)
/* For the nRF51-based QEMU Cortex-M0 platform, the first set of consecutive
 * implemented interrupts that can be used by this test starts right after
 * the TIMER0 IRQ line, which is used by the system timer.
 */
#define _ISR_OFFSET (TIMER0_IRQn + 1)
#elif defined(CONFIG_SOC_SERIES_NRF54LX)
/* For nRF54L Series, use SWI00-02 interrupt lines. */
#define _ISR_OFFSET SWI00_IRQn
#else
/* For other nRF targets, use TIMER0-2 interrupt lines. */
#define _ISR_OFFSET TIMER0_IRQn
#endif
#endif /* CONFIG_SOC_FAMILY_NRF */
```